internal/pkg/core/authority/ca.go

```diff
@@ -96,6 +100,23 @@ func NewCA() (*CertificateAuthority, error) {
 	}, nil
 }
 
+func generateSubjectKeyID(pub crypto.PublicKey) []byte {
```

Comment:
Generating our own subject key like this avoids the default SHA1 id
Copied from https://github.com/elastic/elastic-agent-libs/pull/263

Reply:
It seems a bit risky to me to copy the implementation over, as implementations could diverge between here and elastic-agent-libs.  Could we not export this function in elastic-agent-libs and call it from here instead?

Reply:
Most of the time this function is just used in test code